packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -398,13 +426,44 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     }
   }
 
+  void _calculateMaxHeight() {
+    final RenderBox? renderBox = context.findRenderObject() as RenderBox?;
+    setState(() {
+      _maxHeight = renderBox?.size.height ?? 0.0;
+    });
+  }
+
+  void _onScroll() {
+    if (_ancestorScrollPosition == null) {
```

Comment:
You should check if the `pixels` property can be used, with `hasPixels`. The scroll position might not be attached.